gammapy/scripts/analysis.py

```diff
@@ -51,6 +60,124 @@ def settings(self):
+        self.config.validate()
+
+        datastore_path = make_path(self.settings["observations"]["datastore"])
+        if datastore_path.is_file():
+            self.datastore = DataStore().from_file(datastore_path)
+        elif datastore_path.is_dir():
+            self.datastore = DataStore().from_dir(datastore_path)
+        else:
+            log.error("Datastore {} not found.".format(datastore_path))
+            return False
+
+        ids = set()
+        selection = dict()
+        for criteria in self.settings["observations"]["filter"]:
+            selected_obs = ObservationTable()
+
+            selection["type"] = criteria["filter_type"]
+            if "inverted" in criteria and criteria["inverted"]:
+                selection["inverted"] = True
+            if selection["type"] == "sky_circle":
+                selection["frame"] = criteria["frame"]
+                selection["lon"] = Angle(criteria["lon"])
+                selection["lat"] = Angle(criteria["lat"])
+                selection["radius"] = Angle(criteria["radius"])
+                selection["border"] = Angle(criteria["border"])
+            if selection["type"] == "par_box":
+                selection["variable"] = criteria["variable"]
+                selection["value_range"] = criteria["value_range"]
+            if selection["type"] == "angle_box":
+                selection["type"] = "par_box"
+                selection["variable"] = criteria["variable"]
+                selection["value_range"] = Angle(criteria["value_range"])
+
+            if selection["type"] == "sky_circle" or selection["type"].endswith("_box"):
+                selected_obs = self.datastore.obs_table.select_observations(selection)
+            if selection["type"] == "par_value":
+                mask = (
+                    self.datastore.obs_table[criteria["variable"]]
+                    == criteria["par_value"]
+                )
+                selected_obs = self.datastore.obs_table[mask]
+            if selection["type"] == "ids":
+                obs_list = self.datastore.get_observations(criteria["obs_ids"])
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            if selection["type"] == "all":
+                obs_list = self.datastore.get_observations()
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            if len(selected_obs):
+                if "exclude" in criteria and criteria["exclude"]:
+                    ids.difference_update(selected_obs["OBS_ID"].tolist())
+                else:
+                    ids.update(selected_obs["OBS_ID"].tolist())
+        if len(ids):
+            self.observations = self.datastore.get_observations(ids)
+        else:
+            self.observations = Observations()
+
+    def reduce_data(self):
+        """Produce reduced data sets."""
+        self.config.validate()
+
+        # create geometry
+        self.geom = WcsGeom.create(
```

Comment:
I think data reduction has to take either the 1D or 3D path, i.e. decide and have different methods for those.
For 1D no WCS object is needed, that should be in the 3D path (and probably not in this PR)?

Reply:
umm.. in principle I agree, but I think I need the geometry to create an eventual exclusion mask that will be (un-)considered in the reflected regions methof to estimate the background. is it possible to create an exclusion mask without a geometry?

Reply:
Exclusion mask is difficult. You'd also have to specify config which regions to exclude.
Suggest to take an exclusion mask FITS filename, optionally.
Whether to later also allow config to create exclusion regions I'm not sure.

Of course, for 3D will will need to create a geom from config, so don't throw the code away.
I don't know yet whether we want to limit the use of config to Analysis, or use it throughout Gammapy, i.e. in this case there could be a MapGeom.from_config(config_dict) that takes this part of the config. Maybe develop that in utility functions in analysis.py for now?
I probably won't be able to join the dev call tomorrow, this is something you should also bring up as a question and gather more opinions there.

Reply:
Ok, I've put it the *build geometry* part aside as a helper function and deal with exclusion_mask as a FITS file.